### PR TITLE
New feature : About dialog

### PR DESCRIPTION
**Pull request by LauraRozier**
I thought that the plugin was missing a small 'about this plugin' dialog, this PR adds one to the plugin.

### CODE REVIEW

File: src/URLCtrl.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00,
+};
+
+
+void URLCtrl::create(HWND itemHandle, LPTSTR link, COLORREF linkColor)
+{
+	// turn on notify style
+    ::SetWindowLong(itemHandle, GWL_STYLE, ::GetWindowLong(itemHandle, GWL_STYLE) | SS_NOTIFY);
```
Comment (chcg) [inline src/URLCtrl.cpp]: x64 requires SetWindowLongPtr and GetWindowLongPtr, etc.

See https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/WinControls/AboutDlg/URLCtrl.cpp

Seems you took an older version before N++ was prepared for x64

**Comment by LauraRozier** [comment]
True, forgot to push that commit.
Re-did the edits at home.

File: include/URLCtrl.h [hunk 2]
```diff
@@ -21,19 +21,18 @@
 #include <windows.h>
 #include "Window.h"
```
Comment (LauraRozier) [inline include/URLCtrl.h]: Also added a few cleanups.

File: src/URLCtrl.cpp [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+		_tcscpy(_URL, link);
+
+	// set the hyperlink colour
+    _linkColor = linkColor;
+
+	// set the visited colour
+	_visitedColor = RGB(128,0,128);
+
+	// subclass the static control
+	_oldproc = (WNDPROC)::SetWindowLongPtr(itemHandle, GWLP_WNDPROC, (LONG)URLCtrlProc);
```
Comment (chcg) [inline src/URLCtrl.cpp]: Think you could take directly the version from the current n++ trunk or did you do any modification to this file. 

Here and below the cast to LONG is not correct needs to be LONG_PTR, see n++ version.

File: src/URLCtrl.cpp [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+	// associate the URL structure with the static control
+	::SetWindowLongPtr(itemHandle, GWLP_USERDATA, (LONG)this);
```
Comment (chcg) [inline src/URLCtrl.cpp]: Here too (LONG) needs to be reinterpret_cast<LONG_PTR>()

**Comment by LauraRozier** [comment]
Odd, thought I got it from the correct source, seems I was digging around in a very old repo.
Got new ones from their GH repo.

**Pull request merged**